PS C:\Users\B730369\aws-reports\lib\lambda\test> python -m pytest patching_report_test.py
====================================================== test session starts ======================================================
platform win32 -- Python 3.10.12, pytest-8.3.4, pluggy-1.5.0
rootdir: C:\Users\B730369\aws-reports
configfile: pytest.ini
collected 2 items

patching_report_test.py F.                                                                                                 [100%]

=========================================================== FAILURES ============================================================ 
__________________________________________________ test_lambda_handler_success __________________________________________________ 

mock_s3_client = <MagicMock name='client()' id='1811753227584'>

    def test_lambda_handler_success(mock_s3_client):
        # Mock input and output bucket names
        input_bucket = "op-cloudcustodian-reports"
        output_bucket = "op-patching-reports"
        output_key = "csv/instance-patching-report.csv"

        # Mock event and context
        event = {}
        context = {}

        # Call the Lambda handler
        response = lambda_handler(event, context)

        # Assertions
>       assert response["statusCode"] == 200
E       assert 500 == 200

patching_report_test.py:51: AssertionError
----------------------------------------------------- Captured stdout call ------------------------------------------------------ 
Error: 'str' object has no attribute 'decode'
==================================================== short test summary info ==================================================== 
FAILED patching_report_test.py::test_lambda_handler_success - assert 500 == 200
================================================== 1 failed, 1 passed in 0.90s ================================================== 
PS C:\Users\B730369\aws-reports\lib\lambda\test> python -m pytest patching_report_test.py
====================================================== test session starts ======================================================
platform win32 -- Python 3.10.12, pytest-8.3.4, pluggy-1.5.0
rootdir: C:\Users\B730369\aws-reports
configfile: pytest.ini
collected 2 items

patching_report_test.py F.                                                                                                 [100%]

=========================================================== FAILURES ============================================================ 
__________________________________________________ test_lambda_handler_success __________________________________________________ 

mock_s3_client = <MagicMock name='client()' id='1815153104000'>

    def test_lambda_handler_success(mock_s3_client):
        # Mock input and output bucket names
        input_bucket = "op-cloudcustodian-reports"
        output_bucket = "op-patching-reports"
        output_key = "csv/instance-patching-report.csv"
    
        # Mock event and context
        event = {}
        context = {}

        # Call the Lambda handler
        response = lambda_handler(event, context)

        # Assertions
        assert response["statusCode"] == 500
>       assert "Report successfully saved" in response["body"]
E       assert 'Report successfully saved' in "Error: 'str' object has no attribute 'decode'"

patching_report_test.py:52: AssertionError
----------------------------------------------------- Captured stdout call ------------------------------------------------------ 
Error: 'str' object has no attribute 'decode'
==================================================== short test summary info ==================================================== 
FAILED patching_report_test.py::test_lambda_handler_success - assert 'Report successfully saved' in "Error: 'str' object has no attribute 'decode'"
